### PR TITLE
fix #2968

### LINKED ISSUE #2968: Hundreds wrong detection on yolov8
Using code writed for yolov5, after some adjustment for yolov8, when I try to use torchscript model generated I got ~200 detection with very high score (10000+), and they are mostly focused in left top corner. My model probably is created correctly using yolov8s/yolov8n model (Ultralytics tool) and `yolo predict` work great on that model. Then I exported to torchscript format (`yolo export model=... format=torchscript` like v5 version).

1-Figure1-1

I updated djl to 0.26 and changed translator to `YoloV8Translator`.
My yolo usage looks like:
```
        Image picture = ImageFactory.getInstance().fromFile(image.toPath());
        
        Pipeline pipeline = new Pipeline();
        pipeline.add(new Resize(640));
        pipeline.add(new ToTensor());

        List<String> synset = new ArrayList<>(4);
        synset.add(DETECTION_LABEL_0);
        synset.add(DETECTION_LABEL_1);
        synset.add(DETECTION_LABEL_2);
        synset.add(DETECTION_LABEL_3);

        YoloV8Translator translator = YoloV8Translator
                .builder()
                .setPipeline(pipeline)
                .optSynset(synset)
                .optThreshold(0.7f)
                .build();

        Criteria criteria = Criteria.builder()
                .setTypes(Image.class, DetectedObjects.class)
                .optDevice(Device.cpu())
                .optModelPath(Paths.get(modelLocation))
                .optTranslator(translator)
                .optEngine("PyTorch")
                .build();

        ZooModel<Image, DetectedObjects> model = criteria.loadModel();
        Predictor<Image, DetectedObjects> predictor = model.newPredictor();
        predictor.predict(picture);
```

Same code with YoloV5Translator, djl 0.19.0 and model built on yolov5 works fine.
What I forgot, or what I do wrong? Can you help me migrate to v8?

### PR DESCRIPTION
**Pull request by SidneyLann**
fix #2968

### CODE REVIEW

File: api/src/main/java/ai/djl/modality/cv/translator/YoloV8Translator.java [hunk 1]
```diff
@@ -75,7 +75,7 @@ protected DetectedObjects processFromBoxOutput(NDList list) {
                 float classProb = buf[index + c];
                 if (classProb > maxClassProb) {
                     maxClassProb = classProb;
-                    maxIndex = c;
+                    maxIndex = c - 4;
```
Comment (frankfliu) [inline api/src/main/java/ai/djl/modality/cv/translator/YoloV8Translator.java:78]: maxIndex = c is corret.

nClasses should be the same as classes.length. We might add a check to ensure we provide correct error message.

**Review by frankfliu — CHANGES_REQUESTED** [review]

File: api/src/main/java/ai/djl/modality/cv/translator/YoloV8Translator.java (hunk 1 above)
Comment (SidneyLann) [reply api/src/main/java/ai/djl/modality/cv/translator/YoloV8Translator.java:78]: No. maxIndex should start with 0, but not 4!

Comment (SidneyLann) [reply api/src/main/java/ai/djl/modality/cv/translator/YoloV8Translator.java:78]: The labels 0, 1, 2, 3 should never be detected for the bug.

Comment (SidneyLann) [reply api/src/main/java/ai/djl/modality/cv/translator/YoloV8Translator.java:78]: For the existing logic, the max value of maxIndex is classes.length + 3, but not classes.length -1

Comment (SidneyLann) [reply api/src/main/java/ai/djl/modality/cv/translator/YoloV8Translator.java:78]: Or, for (int c = 4; c < nClasses; c++) should change to for (int c = 0; c < nClasses; c++)?

Comment (SidneyLann) [reply api/src/main/java/ai/djl/modality/cv/translator/YoloV8Translator.java:78]: maybe this is correct:
            for (int c = 0; c < nClasses; c++) {
                float classProb = buf[index + 4 + c];
                if (classProb > maxClassProb) {
                    maxClassProb = classProb;
                    maxIndex = c;
                }
            }

Comment (frankfliu) [reply api/src/main/java/ai/djl/modality/cv/translator/YoloV8Translator.java:78]: I see where is the problem. Original COCO dataset has 80 classes, but in synset.txt contains 84 rows and first 4 are dummy:

0 = "# Classes for coco dataset on which yelov8 is trained"
1 = "# source config https://github.com/ultralytics/ultralytics/blob/main/ultralytics/cfg/datasets/coco.yaml."
2 = "# COCO dataset website: https://cocodataset.org/#home"
3 = "# Ultralytics Coco doc page: https://docs.ultralytics.com/datasets/detect/coco/"

So if you use YoloV8Translator, you need pad 4 dummy labels.

**Review by frankfliu — APPROVED** [review]

File: api/src/main/java/ai/djl/modality/cv/translator/YoloV8Translator.java (hunk 1 above)
Comment (SidneyLann) [reply api/src/main/java/ai/djl/modality/cv/translator/YoloV8Translator.java]: if (padding != 0 && padding != 4) { //means padding ==0 or padding ==4 is invalid, this is not true. the check should be removed and change for (int c = 4; c < nClasses; c++) { to for (int c = padding; c < nClasses; c++) {?

Comment (SidneyLann) [reply api/src/main/java/ai/djl/modality/cv/translator/YoloV8Translator.java]: if (padding != 0 && padding != 4) { //means padding ==0 or padding ==4 is invalid, this is not true. the check should be removed and change for (int c = 4; c < nClasses; c++) { to for (int c = padding; c < nClasses; c++) {?

**Comment by SidneyLann** [comment]
The merging logic is not correct

**Comment by frankfliu** [comment]
The output shape is (8400, 84) for COCO dataset, we should always ignore first 4 entries in the output.
Ideally, we should not padding classes with dummy classes, the originally implementation is a hack. Keep the 4 padding behavior is purely for backward compatibility. We should not allow other paddings.

**Comment by SidneyLann** [comment]
But 0 padding or 4 padding is not work now

**Comment by frankfliu** [comment]
0 or 4 should work, it throw exception when not 0 **and** not 4
```
if (padding != 0 && padding != 4) {
```

**Comment by SidneyLann** [comment]
If padding is 0, the min label is 4, it should be 0

**Comment by frankfliu** [comment]
I think the variable name `padding` is misleading. It should `diff`,

if `diff` == 0, it mean the classes has already padded with dummy entry, so the index should as is.
If `diff` == 4, it means the classes doesn't have padding, so we should -4 to adjust the index

**Comment by SidneyLann** [comment]
I finetune yolov8 in my dataset and get a new yolov8 model for 96 classes, there are 96 entries in my synset.txt and has no dummy  entries. now the diff is 4(doesn't have padding)? 

int padding = nClasses - classes.size();// nClasses is from synset.txt and equals to 96? classes.size() is from the model and equals to 96? 

So the padding(diff) is 0?

**Comment by frankfliu** [comment]
If you trained your model with 96 classes, the output should should be (8400, 100)

`nClasses - classes.size()` should be 4
